### ISSUE TITLE
WO-022: delivery timing (stochastic, backloaded)

### ISSUE THREAD

**Issue by Xocas12** [Xocas12/gosplan-env#27]
> ## NOT ISSUABLE YET - THIS CARD IS A PLACEHOLDER AND ITS CONTENT IS PROVISIONAL
>
> **This work order is issued only AFTER the Phase-2 spec revision.** It is gated on gate **G2** and
> on the lead's spec revision that follows G2. Nothing here may be started before both have happened.
>
> **The behaviour of the P2 sketch is frozen at that revision, and not before.** PLAN §12.4 gives
> WO-022 as the summary "delivery timing (`stochastic`, `backloaded`)"; PLAN §2.6 gives one sentence
> of mechanism and PLAN §4.2 gives the locked parameter values. That is all that is decided today.
>
> **Everything under "Implementation notes" below is PROVISIONAL** until the lead revises the spec.
> Do not read it as settled and do not present it to anyone else as settled.
>
> **An implementer handed this card in its present form stops and files an AMBIGUITY REPORT**
> (`workorders/AMBIGUITY_TEMPLATE.md`, CONTRACT rule 3) instead of choosing a reasonable default.

> **WO-022** - Phase 2 - Owner **MID-fast** - Difficulty **set by the LEAD at issue** (PLAN §12.4 fixes the tier only) - Gate **G3**
> Card: [`workorders/WO-022.md`](https://github.com/Xocas12/gosplan-env/blob/main/workorders/WO-022.md)

## What this is

Makes delivered inputs arrive *during* a period instead of all at once at its start. Today every
delivery lands at step 0 and an enterprise can spread its effort evenly across the period's PRODUCE
steps; with `delivery_timing = "stochastic"` each unit of delivery arrives at a step drawn from a
categorical distribution, so the inputs needed to produce may simply not be there yet. That is the
physical precondition for storming - end-of-period effort spikes - without anything in the code
telling an agent to defer effort. A separate `backloaded` setting exists for a late-weighted arrival
distribution; it is never the acceptance value.

## Depends on

- gate G2 - the Phase-1 gate (PLAN §4.5 criteria 1-4, signed off Human + LEAD). This card cannot be
  started before G2 has passed.
- the **P2 spec revision** (LEAD) - the lead's post-G2 revision, which is what freezes the
  behaviour described below.
- WO-013
- WO-004
- WO-005
- WO-006
- WO-009

Consumed by WO-030 (phenomenon row 2, storming) and WO-031 (the P2 acceptance run).

**Blocked by:** #18 (WO-013), #9 (WO-004), #10 (WO-005), #11 (WO-006), #14 (WO-009), #45 (gate G2)
This card may not be started until every issue above is closed (CONTRACT rule 12 / the loop in CONTRIBUTING.md).

## Read first (exhaustive whitelist)

- `PLAN.md` §§ 2.5, 2.6, 2.7.3, 2.15, 3, 4.2, 4.4 - PLAN.md is git-ignored and supplied out of band;
  it is not in the repository and has no blob link.
- [`CONTRACT.md`](https://github.com/Xocas12/gosplan-env/blob/main/CONTRACT.md) - rules 7, 9
- [`spec/spec.py`](https://github.com/Xocas12/gosplan-env/blob/main/spec/spec.py) - `DeliveryTiming`,
  `SupplyConfig.delivery_timing`, `.arrival_probs`, `Purpose`, `draw`, `produce_step`
- [`spec/CHANGELOG.md`](https://github.com/Xocas12/gosplan-env/blob/main/spec/CHANGELOG.md)
- [`gosplan/config.py`](https://github.com/Xocas12/gosplan-env/blob/main/gosplan/config.py)
- [`gosplan/params.py`](https://github.com/Xocas12/gosplan-env/blob/main/gosplan/params.py)
- [`gosplan/rng.py`](https://github.com/Xocas12/gosplan-env/blob/main/gosplan/rng.py)
- [`gosplan/env/production.py`](https://github.com/Xocas12/gosplan-env/blob/main/gosplan/env/production.py)
- [`gosplan/env/planner.py`](https://github.com/Xocas12/gosplan-env/blob/main/gosplan/env/planner.py)
- [`gosplan/env/state.py`](https://github.com/Xocas12/gosplan-env/blob/main/gosplan/env/state.py)
- [`gosplan/env/step.py`](https://github.com/Xocas12/gosplan-env/blob/main/gosplan/env/step.py)
- [`docs/ref_worked_example.md`](https://github.com/Xocas12/gosplan-env/blob/main/docs/ref_worked_example.md)
- **the revised spec text, which does not exist yet - that is what gates this card**

Read nothing else. CONTRACT rule 12.

## Write only

Provisional; the revision fixes the final list.

- `gosplan/env/planner.py` - the arrival schedule attached to `deliv_bj` at DELIVER
- `gosplan/env/state.py` - the in-flight arrival buffer, if the revision adds one
- `gosplan/env/production.py` - the point at which a scheduled unit becomes available to `coverage`

Not this card: `ref/`, `tests/**`, `spec/spec.py`, `gosplan/metrics/phenomena.py`. Golden-file
regeneration and any `spec/CHANGELOG.md` entry are LEAD work (CONTRACT rules 1, 2).

Creating, editing or deleting any other file is a CONTRACT rule 12 violation.

## Must pass

**No new test ids are frozen yet** - the revision writes them and names their ids in this section
(CONTRACT rule 12).

Already binding and already determinate, green after this change:

- [ ] **T-U1** per-period per-good conservation to 1e-9, with units in flight accounted for
      explicitly - `tests/unit/test_conservation.py`
- [ ] **T-B7** golden parity with `ref/` - `tests/golden/test_golden_parity.py`
- [ ] **T-B1** no hard-coded pathology: under `delivery_timing = "uniform"` the within-period effort
      Gini of `TruthfulMyopic` equals that implied by yield noise alone -
      `tests/behavioural/test_no_hardcoded_pathology.py`

Acceptance criterion PLAN already fixes: **at `delivery_timing = "uniform"` (the Phase-1 default)
everything arrives at step 0 and this card's code path is the exact identity**, so every golden file
passes unchanged.

Completion command, copy-paste:

```bash
uv run ruff check . && uv run ruff format --check . && uv run pytest -q
```

The revision appends the specific `pytest tests/unit/<new test file> -q` invocation. `pytest -q`
never collects `tests/acceptance/` (CONTRACT rule 13).

## Implementation notes

**Provisional until the lead revises the spec.** What follows is only what PLAN.md already
determines.

### Locked Phase-2 mechanism parameters (PLAN §4.2, row "2 Storming")

Locked **now**; not this card's to change, nor WO-030's, nor WO-031's:

```
delivery_timing = "stochastic"
arrival_probs   = π_arrival = (0.25, 0.25, 0.25, 0.25)   over steps 0..3
yield_sigma     × 1 (the Phase-1 tuple, unchanged)
```

PLAN §4.2 states why the arrival distribution is uniform-random: there is then **no mechanical
backloading**, so any effort-Gini excess measured later is behavioural rather than an artefact of
when inputs happened to show up. Making the arrival distribution late-weighted in the acceptance
configuration would manufacture the very phenomenon the study holds out. `backloaded` exists as a
separate configuration, never as the acceptance value.

### The draw

A delivered unit's arrival step is

```
k ~ Categorical(π_arrival),  π_arrival = supply.arrival_probs
```

drawn through `gosplan.rng.draw(seed_env, "arrival", …)`. Purpose `arrival` is already in the
`Purpose` enum of `spec/spec.py` (PLAN §2.15). CONTRACT rule 9: no `numpy.random` and no
`jax.random` under `gosplan/env/`, ever.

Keying must keep the draw **order-independent** - PLAN §2.15 keys every draw as
`SeedSequence([seed_env, crc32(purpose), *indices])`, not by walking a stream. Two consequences that
are requirements, not niceties:

- the JAX port of WO-029 reproduces this draw exactly (its parity criterion is 100 agent-steps
  identical to NumPy to 1e-5), which is only possible if the draw does not depend on call order;
- common random numbers hold across arms, which is what makes the WO-030 row-2 comparison against
  the truthful-myopic baseline under the *same* delivery-timing draws a valid comparison rather than
  a noise draw.

### Sign conventions, units, edge cases

- Arrival steps are indices into the period's PRODUCE steps, `0 .. steps_per_period − 1`. `π_arrival`
  is a probability vector over those steps and must sum to 1.
- `delivery_timing = "uniform"` means everything arrives at step 0. It is the Phase-1 default and
  must be an explicit early branch that does no drawing at all, so T-B7 golden parity holds
  bit-for-bit and the RNG stream is untouched.
- Conservation (T-U1, tolerance **1e-9** per period per good): units in flight are still units. A
  quantity that has been shipped but has not yet arrived must appear as an explicit term of the
  per-period per-good identity; it must not vanish from the accounting between DELIVER and its
  arrival step, and it must not be double-counted at both ends.
- `EnvConfig.validate` requires `len(arrival_probs) == steps_per_period` when
  `delivery_timing != "uniform"`.

### What PLAN does NOT decide - the revision must state each before this card is issued

- **the index set of the draw** - one arrival step per (buyer, good) delivery, per unit, or per
  (buyer, good, seller-pool). This changes the variance of coverage within a period, and therefore
  changes the measured storming excess. It is not a free choice.
- **`steps_per_period` other than 4.** `EnvConfig.validate` requires
  `len(arrival_probs) == steps_per_period` when `delivery_timing != "uniform"`; at
  `steps_per_period = 8` (in the PLAN §3 grid) the locked 4-tuple no longer applies and the revision
  must say what does. Do not pad, truncate or renormalise the locked tuple to fit.
- **the `backloaded` distribution itself.** "π concentrated on late steps" is not a number.
- **whether goods arriving at step `k` are usable at step `k` or only at `k + 1`.** This is the
  difference between arrival relieving the current step's coverage constraint and relieving the next
  one's, and it moves the effort profile directly.

Do not pick any of these. CONTRACT rule 3: choosing "the reasonable default" is a violation.

## Forbidden

- Computing, plotting or testing PLAN §4.1 rows **2, 5, 6, 7** - held out. Row 2 (storming) is this
  mechanism's own phenomenon and is computed for the first time in WO-030, inside the P2 acceptance
  run of WO-031 only.
- Any rule that defers effort, concentrates effort late in a period, or rewards doing so. Storming
  must be a consequence of the arrival process and the reward (CONTRACT rule 7).
- Direct `numpy.random` / `jax.random` under `gosplan/env/` (CONTRACT rule 9).
- Changing `π_arrival`, `delivery_timing` or `yield_sigma` away from the PLAN §4.2 locked values.
  Those may be changed only in a new, labelled study with its own pre-registration.
- Editing `spec/spec.py`, `tests/**` or `ref/`, or regenerating golden files (CONTRACT rules 1, 2).

## Contract rules that bite here

- **Rule 3 - STOP AND REPORT.** Four questions above are undecided, and the index set of the draw in
  particular changes the measured phenomenon. Choosing "the reasonable default" is a violation.
- **Rule 12 - work orders.** Read only the whitelist, write only the listed files, run the
  completion command verbatim, report in the four-part format.
- **Rule 9 - RNG.** The arrival draw goes through `rng.draw(seed_env, "arrival", *indices)`,
  order-independently. This is what makes JAX parity and common random numbers possible.
- **Rule 7 - no hard-coded pathology.** Nothing here may defer or concentrate effort.
- **Rule 2 - frozen tests.** T-B1's Gini clause is a frozen test; if it looks wrong, file an
  ambiguity report, do not special-case the implementation to pass it.
- **Rule 13 - tests are not experiments.** `tests/acceptance/` is never run from this session.

## Definition of done

- [ ] Every must-pass test green, including the frozen Phase-1 suite unchanged
- [ ] At `delivery_timing = "uniform"` every golden file passes unchanged
- [ ] Units in flight appear explicitly in the T-U1 conservation identity
- [ ] `uv run ruff check . && uv run ruff format --check .` clean
- [ ] `uv run python scripts/contract_guard.py --all` clean
- [ ] No file outside "Write only" created, edited or deleted
- [ ] CI `ci-ok` green on the PR
- [ ] The four-part report posted as the PR body: 1) tests passed/failed with output, 2) ambiguity
      reports filed, 3) files changed, 4) anything you were tempted to invent and did not
